src/missingarray.jl

```diff
@@ -26,3 +26,13 @@ Base.fill!(A::CategoricalArray{>:Missing}, ::Missing) = (fill!(A.refs, 0); A)
 
 in(x::Missing, y::CategoricalArray) = false
 in(x::Missing, y::CategoricalArray{>:Missing}) = !all(v -> v > 0, y.refs)
+
+function Missings.replace(a::CategoricalArray{S, N, R, V, C}, replacement::V) where {S, N, R, V, C}
+    pool = CategoricalPool(a.pool)
```

Comment:
Don't forget to change this to `copy(a.pool)`.

Reply:
`copy` isn't implemented. `deepcopy` works though. But I don't think a `deepcopy` is in general preferred over a shallow copy here.

Reply:
What we really need is a deepcopy here, so that's fine.

Reply:
ah, `CategoricalPool(a.pool)` is to shallow, changing to deepcopy